### PR TITLE
chore: don't wait for e2e tests if only changing the plugin server

### PR DESCRIPTION
**Pull request by pauldambra**
If you only change the plugin server you spend a long time waiting for e2e CI to run.

It doesn't use the plugin server (I don't think)

So, don't run it...

### CODE REVIEW

File: .github/workflows/ci-e2e.yml [hunk 1]
```diff
@@ -135,12 +173,15 @@ jobs:
                   $DOCKER_RUN ./bin/migrate
                   $DOCKER_RUN python manage.py setup_dev
 
+                  # only starts the plugin server so that the "wait for PostHog" step passes
                   $DOCKER_RUN ./bin/docker-worker &> /tmp/logs/worker.txt &
```
Comment (pauldambra) [inline .github/workflows/ci-e2e.yml:177]:
```suggestion
                  $DOCKER_RUN ./bin/docker-worker &> /tmp/logs/worker.txt &
                  $DOCKER_RUN ./bin/docker-server &> /tmp/logs/server.txt &
```

**Comment by pauldambra** [comment]
@hazzadous @benjackwhite when we change only plugin server code it is a smidge frustrating to then wait 20 odd minutes for Cypress tests.

I've confirmed locally that they pass without the plugin server running. So I don't think we can break the Cypress tests with changes to the plugin server.

So, I propose we don't run them unless we've changed backend or frontend code (which I think this PR does)

File: .github/workflows/ci-e2e.yml [hunk 2]
```diff
@@ -13,8 +13,46 @@ concurrency:
     cancel-in-progress: true
 
 jobs:
+    changes:
+        runs-on: ubuntu-latest
+        timeout-minutes: 5
+        if: github.repository == 'PostHog/posthog'
+        name: Determine need to run E2D checks
```
Comment (pauldambra) [inline .github/workflows/ci-e2e.yml:20]:
```suggestion
        name: Determine need to run E2E checks
```

File: .github/workflows/ci-e2e.yml [hunk 3]
```diff
@@ -13,8 +13,46 @@ concurrency:
...
+                        # the dependencies more clear if we separated the backend/frontend
+                        # code completely
+                        - 'ee/**/*'
+                        - 'posthog/**/*'
+                        - 'bin/*.py'
+                        - requirements.txt
+                        - requirements-dev.txt
+                        - mypy.ini
+                        - pytest.ini
+                        - frontend/src/queries/schema.json # Used for generating schema.py
```
Comment (benjackwhite) [inline .github/workflows/ci-e2e.yml:44]: Is this not covered by `frontend/**/*` ?

**Review by benjackwhite — APPROVED** [review]

File: .github/workflows/ci-e2e.yml (hunk 3 above)
Comment (pauldambra) [inline .github/workflows/ci-e2e.yml:44]:
```suggestion
```

**Comment by hazzadous** [comment]
We do use the plug-in server for cypress tests I believe. If you wanted to not run them on blob store tests I’d maybe look at running it separately from the plug-in servrr

**Comment by hazzadous** [comment]
But also yeah we could just not run them. It seems they rarely are related.

I’ve changed my mind down with cypress tests on plug-in server changes!

**Review by hazzadous — APPROVED** [review]
I feel like eg the e2e preflight test is going to fail now but I’m for not running e2e on plug-in server changes

**Comment by pauldambra** [comment]
> I feel like eg the e2e preflight test is going to fail now but I’m for not running e2e on plug-in server changes

@hazzadous good call! I stubbed the _preflight responses for that test and added a test that we show errors if there are any

**Comment by hazzadous** [comment]
🥳

**Pull request merged**